### ISSUE TITLE
`BitVec::resize` unexpectedly panics

### ISSUE THREAD

**Issue by jonas-schievink** [myrrlyn/bitvec#33]
Reproduction:
```rust
use bitvec::{order::Lsb0, vec::BitVec, slice::BitSlice};

fn main() {
    let mut dir = BitVec::<Lsb0, u8>::new();
    let mut swdio = BitVec::<Lsb0, u8>::new();

    swdio.resize(64, true);
    dir.resize(64, true);

    let mut seq = 0xE79E; // LSb first
    for _ in 0..16 {
        swdio.push(seq & 0b1 != 0);
        seq >>= 1;
    }

    dir.resize(dir.len() + 16, true);

    swdio.resize(swdio.len() + 64, true);
    dir.resize(dir.len() + 64, true);

    // idle cycles = host clocks interface with SWDIO low
    swdio.resize(swdio.len() + 10, false);
    dir.resize(dir.len() + 10, true);
}
```
Since 0.16 (I think?) this panics with:
```
thread 'main' panicked at 'Capacity overflow: 144 overflows allocation size 128', src/vec/api.rs:328:3
```

which is due to this assertion in `set_len`:

https://github.com/myrrlyn/bitvec/blob/65693044c6cecc566b6065edf9d0b8ac397d61d6/src/vec/api.rs#L328-L333

**Comment by myrrlyn** [comment]
WOW this was a rough one to chase down. Thank you for the report. I've published `0.16.1` to crates.io with the fix commit.

**Comment by jonas-schievink** [comment]
Great, thanks for the quick fix!
